### PR TITLE
test: blockaid update version and reenable specs

### PR DESCRIPTION
**Pull request by seaona**
## **Description**
This PR does the following:
- Changes the specs that were going to the test dapp using the ip address as hostname, to use localhost instead ([see issue here](https://github.com/MetaMask/MetaMask-planning/issues/3560))
- Re-enables skipped specs
- Updates the `update-mock-cdn` script, so it adds `"Content-Type": "text/plain"` into the response headers
- Updates the ppom module to the last version (config, stale and stale diff) using the updated script

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28121?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Check ci

## **Screenshots/Recordings**
Example of malicious signature working on e2e environment

https://github.com/user-attachments/assets/2de66e12-ee52-46eb-9835-129aa1370fd0

Encoded mock data before/after content type

https://github.com/user-attachments/assets/9b91ccaa-5b8b-4854-a093-cd11da9f4a7c

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/ppom/ppom-blockaid-alert-simple-send.spec.js [hunk 1]
```diff
@@ -106,11 +107,9 @@ async function mockInfuraWithFailedResponses(mockServer) {
...
 
   await mockServer
-    .forPost()
-    .withJsonBodyIncluding({
-      method: 'debug_traceCall',
-      params: [{ accessList: [], data: '0x00000000' }],
-    })
+    .forGet(
+      'https://static.cx.metamask.io/api/v1/confirmations/ppom/ppom_version.json',
+    )
```
Comment (seaona) [inline test/e2e/tests/ppom/ppom-blockaid-alert-simple-send.spec.js:112]: to cause a warning, it doesn't work anymore the debug_traceCall 500, but we can use this failed request instead

File: test/e2e/mock-cdn/cdn-config.txt
Comment (seaona) [inline test/e2e/mock-cdn/cdn-config.txt:1]: the updated `txt` files are obtained by running the `update-mock-cdn` script. The data is encoded in brotli, taken from making a real request to the prod endpoints, to get the latest data

File: test/e2e/mock-cdn/cdn-stale-diff-res-headers.json [hunk 2]
```diff
@@ -1,3 +1,4 @@
 {
-  "Etag": "W/\"ece7f5f533b8978063633ea5b1f8a0fc\""
+  "Content-Type": "text/plain",
```
Comment (seaona) [inline test/e2e/mock-cdn/cdn-stale-diff-res-headers.json:2]: with this addition, we are now able to see the mocked response in the network tab (see attached video in the description)

File: test/e2e/tests/ppom/ppom-blockaid-alert-erc20-transfer.spec.js [hunk 3]
```diff
@@ -161,14 +157,17 @@ async function mockInfura(mockServer) {
 }
 
 describe('PPOM Blockaid Alert - Malicious ERC20 Transfer @no-mmi', function () {
-  // eslint-disable-next-line mocha/no-skipped-tests
-  it.skip('should show banner alert', async function () {
```
Comment (seaona) [inline test/e2e/tests/ppom/ppom-blockaid-alert-erc20-transfer.spec.js:165]: not skipping  the test anymore as it's now fixed

File: test/e2e/tests/ppom/ppom-blockaid-alert-erc20-transfer.spec.js [hunk 4]
```diff
@@ -181,10 +180,10 @@ describe('PPOM Blockaid Alert - Malicious ERC20 Transfer @no-mmi', function () {
       async ({ driver }) => {
         const expectedTitle = 'This is a deceptive request';
         const expectedDescription =
-          'If you approve this request, you might lose your assets.';
```
Comment (seaona) [inline test/e2e/tests/ppom/ppom-blockaid-alert-erc20-transfer.spec.js:184]: the malicious descriptions have slightly changed so we need to modify them in the specs

File: test/e2e/tests/ppom/ppom-blockaid-alert-erc20-transfer.spec.js [hunk 5]
```diff
@@ -195,20 +194,15 @@ describe('PPOM Blockaid Alert - Malicious ERC20 Transfer @no-mmi', function () {
...
-        const bannerAlertText = await bannerAlertFoundByTitle.getText();
 
-        assert(
-          bannerAlertFoundByTitle,
-          `Banner alert not found. Expected Title: ${expectedTitle} \nExpected reason: transfer_farming\n`,
-        );
-        assert(
-          bannerAlertText.includes(expectedDescription),
-          `Unexpected banner alert description. Expected: ${expectedDescription} \nExpected reason: transfer_farming\n`,
-        );
```
Comment (seaona) [inline test/e2e/tests/ppom/ppom-blockaid-alert-erc20-transfer.spec.js:211]: we already find the element by its text, so no need to assert after

File: test/e2e/mock-cdn/cdn-config.txt
Comment (hjetpoluru) [reply test/e2e/mock-cdn/cdn-config.txt:1]: Thanks @seaona for documenting this information.

**Review by hjetpoluru — APPROVED** [review]
LGTM

File: test/e2e/tests/ppom/ppom-blockaid-alert-simple-send.spec.js [hunk 6]
```diff
@@ -175,29 +179,25 @@ describe('Simple Send Security Alert - Blockaid @no-mmi', function () {
...
+        await driver.waitForSelector({
+          css: '.mm-text--body-md',
+          text: expectedMaliciousDescription,
+        });
       },
     );
   });
 
-  // eslint-disable-next-line mocha/no-skipped-tests
-  it.skip('should show "Request may not be safe" if the PPOM request fails to check transaction', async function () {
```
Comment (seaona) [inline test/e2e/tests/ppom/ppom-blockaid-alert-simple-send.spec.js:200]: not skipping the test anymore as it's now fixed

**Review by chloeYue — APPROVED** [review]

**Comment by chloeYue** [comment]
LGTM !

**Pull request merged**